Sources/Fullscreen/AdConfirmationView/AdConfirmationView.swift

```diff
@@ -0,0 +1,129 @@
+        view.backgroundColor = .bgPrimary
+        view.showsVerticalScrollIndicator = false
+        view.showsHorizontalScrollIndicator = false
+        return view
+    }()
+
+    private lazy var contentView: UIView = UIView(withAutoLayout: true)
+    private lazy var confirmationObjectView: AdConfirmationObjectView = AdConfirmationObjectView(withAutoLayout: true)
+
+    private lazy var completeButton: FooterButtonView = {
+        let button = FooterButtonView(withAutoLayout: true)
+        button.delegate = self
+        return button
+    }()
+
+    private lazy var receiptInfoLabel: Label = {
+        let label = Label(style: .caption, withAutoLayout: true)
+        label.numberOfLines = 0
+        label.textAlignment = .center
+        return label
+    }()
+
+    public weak var delegate: AdConfirmationViewDelegate?
+
+    public var model: AdConfirmationViewModel? {
+        didSet {
+            confirmationObjectView.model = model?.objectViewModel
+            completeButton.buttonTitle = model?.completeButtonText
+            receiptInfoLabel.text = model?.summaryViewModel?.receiptInfo
+
+            setupSummaryView()
+        }
+    }
+
+    override init(frame: CGRect = .zero) {
+        super.init(frame: frame)
+        backgroundColor = .bgPrimary
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+private extension AdConfirmationView {
+    func setup() {
+        addSubview(completeButton)
+        NSLayoutConstraint.activate([
+            completeButton.leadingAnchor.constraint(equalTo: safeAreaLayoutGuide.leadingAnchor),
+            completeButton.trailingAnchor.constraint(equalTo: safeAreaLayoutGuide.trailingAnchor),
+            completeButton.bottomAnchor.constraint(equalTo: safeAreaLayoutGuide.bottomAnchor)
+        ])
+
+        addSubview(scrollView)
+        NSLayoutConstraint.activate([
+            scrollView.topAnchor.constraint(equalTo: safeAreaLayoutGuide.topAnchor),
+            scrollView.leadingAnchor.constraint(equalTo: safeAreaLayoutGuide.leadingAnchor),
+            scrollView.trailingAnchor.constraint(equalTo: safeAreaLayoutGuide.trailingAnchor),
+            scrollView.bottomAnchor.constraint(equalTo: completeButton.topAnchor)
+        ])
+
+        scrollView.addSubview(contentView)
+        contentView.addSubview(confirmationObjectView)
+
+        NSLayoutConstraint.activate([
+            contentView.heightAnchor.constraint(greaterThanOrEqualTo: scrollView.heightAnchor),
+            contentView.widthAnchor.constraint(equalTo: scrollView.widthAnchor),
+            contentView.topAnchor.constraint(equalTo: scrollView.topAnchor),
+            contentView.bottomAnchor.constraint(equalTo: scrollView.bottomAnchor),
+            contentView.leadingAnchor.constraint(equalTo: scrollView.leadingAnchor),
+            contentView.trailingAnchor.constraint(equalTo: scrollView.trailingAnchor),
```

Comment:
I may be mistaken, but I think the lines above could be replaced with `contentView.fillInSuperview()`